src/brsTypes/components/RoSGNode.ts

```diff
@@ -1604,6 +1675,24 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
             }
         });
     }
+
+    /**
+     *  Checks the node sub type hierarchy to see if the current node is a sub compnent of the given node type
+     *
+     * @param {string} checkType
+     * @returns {boolean}
+     */
+    protected isSubtypeCheck(currentNodeType?: string, checkType?: string): boolean {
```

Comment:
In what cases are these values optional?

Reply:
Mostly set as optional in the case where where SubTypeHierarchy.get("someType") returns undefined. :) 
Not really "optional" as a parameter, but more as a convenience.